poseidon-paramgen/src/alpha.rs

```diff
@@ -44,17 +47,26 @@ impl ShortestAdditionChains {
 }
 
 /// The exponent in `Sbox(x) = x^\alpha`.
-#[derive(Clone, Copy, Debug, PartialEq, Eq)]
-pub enum Alpha {
-    /// A positive exponent $x^{alpha}$.
-    Exponent(u32),
-    /// 1/x
-    Inverse,
+#[derive(Debug)]
+pub struct AlphaWrapper(pub Alpha);
+
+impl From<Alpha> for AlphaWrapper {
+    fn from(value: Alpha) -> Self {
+        Self(value)
+    }
+}
+
+impl Deref for AlphaWrapper {
+    type Target = Alpha;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
 }
 
-impl Alpha {
+impl AlphaWrapper {
     /// Select the best choice of `Alpha` given the parameters.
-    pub fn generate<F: PrimeField>(p: F::BigInt, allow_inverse: bool) -> Self {
+    pub fn generate<F: PrimeField>(p: F::BigInt, allow_inverse: bool) -> Alpha {
```

Comment:
Rename this method to `generate_parameters`.

Reply:
I don't think this is a good name:

1. it generates just one parameter (as in singular)
2. again it's better to stick to the same convention, all the other parameters use generate:

Reply:
I left only one comment to show what I propose but actually I propose to change all those names ;)

Reply:
but they all generate just one parameter? if anything than the `PoseidonParametersWrapper` and `InputParametersWrapper` should also have a `generate` (and not "new"). 

Basically *Wrapper is a struct for implementing traits that pull in std (like powers on floats, root calculations or fmt::Debug). They are not "used" in the sense that the entry point to the `paramgen` crate is the `poseidon_build::compile`. Vanilla names are the ones in the upstream `parameters` crate, since these are actually used.

Reply:
My main problem is that the split implicates a need for two struct representations for the same theoretic object, e.g. `Alpha` and `AlphaWrapper` instead of just `Alpha`. The suffix `Wrapper` is a generic one and doesn't contain any info but makes the code look way worse. With my proposition we would have:
  ```
        let input = InputParameters::new(M, t, p, allow_inverse);
        let alpha = Alpha:generate_parameter::<F>(p, allow_inverse);
        let rounds = RoundNumbers::generate_parameter(&input, &alpha);
        let mds = MdsMatrix::generate_parameter(&input);
        let arc = ArcMatrix::generate_parameter(&input, rounds, alpha);
        let optimized_mds = OptimizedMdsMatrices:generate_parameter(&mds, t, &rounds);
        let optimized_arc = OptimizedArcMatrix::generate_parameter(&arc, &mds, &rounds);
```
instead of
```
        let input = InputParametersWrapper::new(M, t, p, allow_inverse);
        let alpha = AlphaWrapper::generate::<F>(p, allow_inverse);
        let rounds = RoundNumbersWrapper::generate(&input, &alpha);
        let mds = MdsMatrixWrapper::generate(&input);
        let arc = ArcMatrixWrapper::generate(&input, rounds, alpha);
        let optimized_mds = OptimizedMdsMatricesWrapper::generate(&mds, t, &rounds);
        let optimized_arc = OptimizedArcMatrixWrapper::generate(&arc, &mds, &rounds);
```
We could even not introduce the `_parameter` suffix. 

Alternatively, instead of `*Wrapper`  call it `*Generator`. Would it make sense?